bugbug/tools/code_review.py

```diff
@@ -517,6 +517,203 @@ def patch_title(self) -> str:
         return self._revision_metadata["fields"]["title"]
 
 
+def create_bug_timeline(comments: list[dict], history: list[dict]) -> list[str]:
+    """Create a unified timeline from bug history and comments."""
+    events = []
+
+    ignored_fields = {"cc", "flagtypes.name"}
+
+    # Add history events
+    for event in history:
+        changes = [
+            change
+            for change in event["changes"]
+            if change["field_name"] not in ignored_fields
+        ]
+        if not changes:
+            continue
+
+        events.append(
+            {
+                "time": event["when"],
+                "type": "change",
+                "who": event["who"],
+                "details": changes,
+            }
+        )
+
+    # Add comments
+    for comment in comments:
+        events.append(
+            {
+                "time": comment["time"],
+                "type": "comment",
+                "who": comment["author"],
+                "id": comment["id"],
+                "count": comment["count"],
+                "text": comment["text"],
+            }
+        )
+
+    # Sort by timestamp
+    events.sort(key=lambda x: (x["time"], x["type"] == "change"))
+
+    # Format timeline
+    timeline = []
+
+    last_event = None
+    for event in events:
+        date = event["time"][:10]
+        time = event["time"][11:19]
+
+        if last_event and last_event["time"] != event["time"]:
+            timeline.append("---\n")
+
+        last_event = event
+
+        if event["type"] == "comment":
+            timeline.append(
+                f"**{date} {time}** - Comment #{event['count']} by {event['who']}"
+            )
+            timeline.append(f"{event['text']}\n")
+        else:
+            timeline.append(f"**{date} {time}** - Changes by {event['who']}")
+            for change in event["details"]:
+                field = change.get("field_name", "unknown")
+                old = change.get("removed", "")
+                new = change.get("added", "")
+                if old or new:
+                    timeline.append(f"  - {field}: '{old}' → '{new}'")
+            timeline.append("")
+
+    return timeline
+
+
+def bug_dict_to_markdown(bug):
+    md_lines = []
+
+    # Header with bug ID and summary
+    md_lines.append(
+        f"# Bug {bug.get('id', 'Unknown')} - {bug.get('summary', 'No summary')}"
+    )
+    md_lines.append("")
+
+    # Basic Information
+    md_lines.append("## Basic Information")
+    md_lines.append(f"- **Status**: {bug.get('status', 'Unknown')}")
+    md_lines.append(f"- **Severity**: {bug.get('severity', 'Unknown')}")
+    md_lines.append(f"- **Priority**: {bug.get('priority', 'Unknown')}")
```

Comment:
Probably not worth having this, since it's not consistent

Reply:
Done here: 94d093c17927e4741a668df04c123731ffa5b82a